OneCommunityGlobal/HighestGoodNetworkApp
# Vijay Anand - Unit test weekly summaries report reducer

# Description
Added unit tests for the weeklySummariesReportReducer.

## Related PRS (if any):
No related PRs.

## Main changes explained:
- Created a new test file weeklySummariesReportReducer.test.js under the tests directory.
- Test cases include:
  - Returns the initial state when no valid action is provided.
  - Handles FETCH_SUMMARIES_REPORT_BEGIN by setting loading to true and clearing the error.
  - Handles FETCH_SUMMARIES_REPORT_SUCCESS by updating the summaries array and setting loading to false.
  - Handles FETCH_SUMMARIES_REPORT_ERROR by updating the error field and setting loading to false.
  - Handles UPDATE_SUMMARY_REPORT by updating the specific summary object in the summaries array.

## How to test:
1. Check out the current branch.
2. Run npm install if necessary.
3. Execute npm test weeklySummariesReportReducer.test.js.
4. Verify that all test cases pass without any errors.

## Screenshots or videos of changes:
<img width="846" alt="image" src="https://github.com/user-attachments/assets/ceadea8b-d14d-4707-90b9-2845e39758c9">

## Note:
Include the information the reviewers need to know.
